src/bridges/mean/MeanBridge.sol

```diff
@@ -151,7 +189,51 @@ contract MeanBridge is BridgeBase, Ownable2Step {
      */
     function computeCriteriaForPosition(address _from, address _to, uint32 _amountOfSwaps, uint32 _swapInterval) public pure returns (uint256) {
         return uint256(keccak256(abi.encodePacked(_from, _to, _amountOfSwaps, _swapInterval)));
-    }    
+    }
+
+    function _wrapAndDeposit(
+        AztecTypes.AztecAsset memory _inputAssetA, 
+        AztecTypes.AztecAsset memory _outputAssetA,
+        uint256 _inputValue,
+        uint64 _auxData
+    ) internal returns(uint256 _positionId, uint256 _criteria) {
+        // Calculate input params
+        (address _from, address _to, uint32 _amountOfSwaps, uint32 _swapInterval) = _mapToPositionData(_inputAssetA, _outputAssetA, _auxData);
+
+        // Wrap input asset if needed
+        uint256 _amountToDeposit = _wrapIfNeeded(_inputAssetA, _from, _inputValue);
+
+        // Make the actual deposit
+        _positionId = DCA_HUB.deposit(
+            _from,
+            _to,
+            _amountToDeposit,
+            _amountOfSwaps,
+            _swapInterval,
+            THIS_ADDRESS,
+            new IDCAHub.PermissionSet[](0)
+        );
+
+        // Compute the criteria for this position
+        _criteria = computeCriteriaForPosition(_from, _to, _amountOfSwaps, _swapInterval);
+    }
+
+    function _wrapIfNeeded(AztecTypes.AztecAsset memory _inputAsset, address _hubToken, uint256 _amountToWrap) internal returns (uint256 _wrappedAmount) {
+        if (_inputAsset.assetType == AztecTypes.AztecAssetType.ETH) {
+            WETH.deposit{value: _amountToWrap}();            
+        } else if (_inputAsset.erc20Address != _hubToken) {
```

Comment:
So, if `inputAsset` is DAI, then `_hubToken` here will be `weDAI`, right? smort 🧠

Reply:
Yes sir!